### ISSUE TITLE
The testCompile configuration is deprecated in Gradle 4

### ISSUE THREAD

**Issue by mttmllns** [allure-framework/allure-gradle#16]
[//]: # (
. Note: for support questions, please use Stackoverflow or Gitter**. 
. This repository's issues are reserved for feature requests and bug reports.
.
. In case of any problems with Allure Jenkins plugin** please use the following repository 
. to create an issue: https://github.com/jenkinsci/allure-plugin/issues
.
. Make sure you have a clear name for your issue. The name should start with a capital 
. letter and no dot is required in the end of the sentence. An example of good issue names:
.
. - The report is broken in IE11
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

#### I'm submitting a ... 
  - [x] bug report

#### What is the current behavior?
When building an Android project using gradle 4.1 and the android gradle plugin 3.0 a warning message is emitted when running any gradle command:

> Configuration 'testCompile' in project ':app' is deprecated. Use 'testImplementation' instead.

The relevant config details:

**gradle/wrapper/gradle-wrapper.properties:**
```properties
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-milestone-1-all.zip
```

**build.gradle:**
```gradle
buildscript {
	dependencies {
		classpath 'com.android.tools.build:gradle:3.0.0-alpha8'
	}
}
```

If the configuration option is manually set to "testImplementation" the warning disappears.

```gradle
allure {
	version = "2.2.1"
	autoconfigure = true
	configuration = "testImplementation"
}
```

#### What is the expected behavior?

Android is using the [new dependency configurations](https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration.html#new_configurations).  It would be nice if this plugin could be cognizant of that setup and use the correct default configuration.

#### Please tell us about your environment:

| Allure version | 2.2.0 |

**Comment by eroshenkoam** [comment]
I will fix in next release, ty for feedback.

**Comment by eroshenkoam** [comment]
@mttmllns why are you using 2.2.0? allure 2.3.5 is already released)

**Comment by mttmllns** [comment]
Didn't notice we were behind. Thanks for the heads up!

**Comment by eroshenkoam** [comment]
@mttmllns I made new version of [allure-gradle](https://github.com/allure-framework/allure-gradle/releases/tag/2.4) few hours ago)

**Comment by mttmllns** [comment]
"SourceSet with name 'test' not found."

Running into this, too.  Looks like it's due to this call:

https://github.com/allure-framework/allure-gradle/blob/master/src/main/groovy/io/qameta/allure/gradle/util/BuildUtils.groovy#L34

I don't believe android projects use the traditional sourceSets so there is no "test" SourceSet. This code should be a little more resilient and check if there's even a "test" sourceset before calling `getByName()`.

**Comment by aikfiend** [comment]
@eroshenkoam, here's a [stacktrace](https://scans.gradle.com/s/yt5c6bcmyu33g/deprecations?expanded=WyI0Il0&expanded-stacktrace=WyI2Il0)

```
plugins {
    id 'java-library'
    id 'io.qameta.allure' version '2.8.1'
}

ext.junitVersion = '5.7.0'
ext.allureVersion = '2.13.8'

allure {
    version = allureVersion
    autoconfigure = true
    aspectjweaver = true
    useJUnit5 {
        version = allureVersion
    }
}
```

Build successful in case of `configuration = "testImplementation"`

**Comment by vlsi** [comment]
Hi, I believe https://github.com/allure-framework/allure-gradle/pull/61 fixes the issue as the plugin no longer adds dependencies to `testCompile`, and it should handle the absence of `test` sourceSet.

Please feel free to try it and provide feedback

**Comment by serpro69** [comment]
Was about to submit a PR for this myself when saw there's already #61 present.
Any plans to merge this soon @eroshenkoam ?

**Comment by vlsi** [comment]
Resolved in #61
